content/posts/lets-code-contact-form-in-phoenix-part-two.md

```diff
@@ -0,0 +1,277 @@
+---
+title: "Let's Code: Contact Form in Phoenix - The Public Interface"
+date: 2020-07-05T00:00:00-05:00
+cover: "/covers/markus-winkler-J2N4mtWcRec-unsplash.jpg"
+coverCredit: "Photo by [Markus Winkler](https://unsplash.com/@markuswinkler?utm_source=unsplash&utm_medium=referral&utm_content=creditCopyText) on [Unsplash](https://unsplash.com/?utm_source=unsplash&utm_medium=referral&utm_content=creditCopyText)"
+draft: true
+comments: false
+tags: ["elixir", "phoenix"]
+---
+
+In this _Let's Code_ series, I will be walking through how I added a simple contact form to a [Phoenix](https://www.phoenixframework.org/) 1.4 web application. I will try to make this as general purpose as possible so that it's easier to follow _and_ apply to your own Phoenix applications. I encourage you to follow along, either in an existing Phoenix application or [a new one](https://hexdocs.pm/phoenix/up_and_running.html#content).
+
+[In Part One]({{< ref "lets-code-contact-form-in-phoenix-part-one" >}}) of this series, I started with a `Message` module to serve as the data object for the contact form. In that post, I described the contact form as taking the minimal number of fields from the user (email, subject, and body) and sending that information to a support email address. In this post, I will be walking through creating a module to send the feedback. Let's get started.
+
+## The Support Interface
+
+For the purposes of a contact form, I know I will need to allow an end user to modify and submit a message to the support team. With that, I can determine two functions: `change_message` and `send_message`. The `change_message` function will be used to represent and validate the data on the form while the `send_message` function will be used to send an email to the support team with the contents of the message. But where do these functions live?
```

Comment:
> I can determine two functions

For some reason "determine" isn't quite clicking in my head.

Reply:
ascertain? make out?

Reply:
I think those both work a bit better for me than "determine" for some reason.
 
My mind keeps thinking of more wordy options like  "expect to create" or"plan to need." Or even more wordier-like: 

With that, I can expect to need two functions: `change_message` (allowing the user to modify the message) and `send_message` (allowing them to send it).